File: Java/프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;

class Solution {
    public int solution(int k, int m, int[] score) {
        HashMap<Integer, Integer> countMap = new HashMap<>();
        for (int item : score) {
            countMap.put(item, (countMap.getOrDefault(item, 0) + 1));
        }
        
        int answer = 0;
        int inBoxRemain = m;
        int inBoxMin = k+1;
        for (int i = k; i>=1; i--) {
            while(countMap.getOrDefault(i, 0) > 0) {
                countMap.put(i,countMap.getOrDefault(i, 0)-1);
                inBoxMin = Math.min(inBoxMin, i);
                inBoxRemain--;
                if (inBoxRemain == 0) {
                    answer += (inBoxMin * m);
                    inBoxRemain = m;
                    inBoxMin = k+1;
                }
            }
        }
        
        return answer;
    }
}

File: Swift/프로그래머스/1/135808. 과일 장수/과일 장수.swift
import Foundation

func solution(_ k: Int, _ m: Int, _ score: [Int]) -> Int {
    let sortedScore = score.sorted(by: { $0 > $1 })
    var count = 0
    var answer = 0
    
    for item in sortedScore {
        count += 1
        if count == m {
            answer += (item * m)
            count = 0
        }
    }
    
    return answer
}
